pkg/yang/types_builtin.go

```diff
@@ -361,24 +300,32 @@ func (r YangRange) Contains(s YangRange) bool {
 // FromInt creates a Number from an int64.
 func FromInt(i int64) Number {
 	if i < 0 {
-		return Number{Kind: Negative, Value: uint64(-i)}
+		return Number{Negative: true, Value: uint64(-i)}
 	}
-	return Number{Kind: Positive, Value: uint64(i)}
+	return Number{Value: uint64(i)}
 }
 
 // FromUint creates a Number from a uint64.
 func FromUint(i uint64) Number {
-	return Number{Kind: Positive, Value: i}
+	return Number{Value: i}
 }
 
 // FromFloat creates a Number from a float64. Input values with absolute value
-// larger than MaxInt64/MinInt64 are converted into maxNumber/minNumber.
+// outside the boundaries specified for the decimal64 value specified in
+// RFC6020/RFC7950 are clamped down to the closest boundary value.
 func FromFloat(f float64) Number {
-	if f > float64(MaxInt64) {
-		return maxNumber
+	if f > MaxDecimal64 {
+		return Number{
+			Value:          FromInt(MaxInt64).Value,
```

Comment:
can you clarify why this is `MaxInt64` rather than whatever was in `f`? It seems like we return a number here that is being treated as magic?